### PR TITLE
Don’t calculate author IDs during initial sync

### PR DESCRIPTION
**Pull request by gravityrail**
cc @ebinnion

### CODE REVIEW

File: sync/class.jetpack-sync-module-full-sync.php [hunk 1]
```diff
@@ -69,6 +69,10 @@ function start( $modules = null ) {
 			$modules = array();
 		}
 
+		if ( isset( $modules['users'] ) && $modules['users'] === 'initial' ) {
```
Comment (gravityrail) [inline sync/class.jetpack-sync-module-full-sync.php]: need to yoda-case this (can do in another PR)

File: sync/class.jetpack-sync-module-full-sync.php [hunk 2]
```diff
@@ -69,6 +69,10 @@ function start( $modules = null ) {
...
+			$modules['users'] = Jetpack_Sync_Actions::get_initial_sync_user_config();
```
Comment (gravityrail) [inline sync/class.jetpack-sync-module-full-sync.php]: This `get_initial_sync_user_config()` function should probably be in the users full sync module. Can do in a separate PR.

File: json-endpoints/jetpack/class.jetpack-json-api-sync-endpoint.php [hunk 3]
```diff
@@ -20,7 +20,9 @@ protected function result() {
 		}
 
 		foreach ( array( 'posts', 'comments', 'users' ) as $module_name ) {
-			if ( isset( $args[ $module_name ] ) ) {
+			if ( 'users' === $module_name && isset( $args[ $module_name ] ) && 'initial' === $args[ $module_name ] ) {
```
Comment (gravityrail) [inline json-endpoints/jetpack/class.jetpack-json-api-sync-endpoint.php]: this special-casing is kind of messy, but I couldn't immediately think of a cleaner way to express it that wasn't overly abstracted.

**Comment by ebinnion** [comment]
This change and your comments make sense to me. To test, I agree that we can get this in now and address your comments later. 🚢

**Comment by gravityrail** [comment]
Merged to 4.2
